### PR TITLE
Fix some inspection hints

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: app/src/main/kotlin/batect/cli/commands/HelpCommand.kt [hunk 1]
```diff
@@ -59,10 +59,7 @@ class HelpCommand(
     private fun nameFor(option: OptionDefinition): String {
         val longNamePart = if (option.acceptsValue) "${option.longOption}=${option.valueFormatForHelp}" else option.longOption
 
-        return when {
-            option.shortName == null -> "    $longNamePart"
-            else -> "${option.shortOption}, $longNamePart"
-        }
+        return option.shortName?.let { "${option.shortOption}, $longNamePart" } ?: "    $longNamePart"
```
Comment (charleskorn) [inline app/src/main/kotlin/batect/cli/commands/HelpCommand.kt:62]: Sorry, this is far less clear to me. Please revert this change.

File: app/src/main/kotlin/batect/cli/options/OptionParser.kt [hunk 2]
```diff
@@ -41,17 +41,11 @@ class OptionParser {
...
-            } else {
-                return OptionParsingResult.NoOption
-            }
+        return optionNames[argName]?.parse(args.drop(currentIndex)) ?: if (argName.startsWith("-")) {
+            OptionParsingResult.InvalidOption("Invalid option '$argName'. Run './batect --help' for a list of valid options.")
+        } else {
+            OptionParsingResult.NoOption
         }
-
-        return option.parse(args.drop(currentIndex))
```
Comment (charleskorn) [inline app/src/main/kotlin/batect/cli/options/OptionParser.kt:54]: Sorry, this is far less clear to me. Please revert this change.

File: app/src/main/kotlin/batect/proxies/ProxyEnvironmentVariablesProvider.kt [hunk 3]
```diff
@@ -34,8 +34,7 @@ class ProxyEnvironmentVariablesProvider(
             lowercaseProxyEnvironmentVariableNames.mapToSet { it.uppercase() }
 
     fun getProxyEnvironmentVariables(extraNoProxyEntries: Set<String>): Map<String, String> {
-        val variables = allPossibleEnvironmentVariableNames
-            .associate { it to hostEnvironmentVariables.getMatchingCaseOrOtherCase(it) }
+        val variables = allPossibleEnvironmentVariableNames.associateWith { hostEnvironmentVariables.getMatchingCaseOrOtherCase(it) }
```
Comment (charleskorn) [inline app/src/main/kotlin/batect/proxies/ProxyEnvironmentVariablesProvider.kt:37]:
```suggestion
        val variables = allPossibleEnvironmentVariableNames
            .associateWith { hostEnvironmentVariables.getMatchingCaseOrOtherCase(it) }
```

File: libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt [hunk 4]
```diff
@@ -45,10 +45,10 @@ data class BasicCredentialsSource(val encodedCredentials: String, val serverAddr
                 throw DockerRegistryCredentialsException("Decoded credentials for '$serverAddress' are not valid.")
             }
 
-            if (parts[0] == "<token>") {
-                return TokenRegistryCredentials(parts[1], serverAddress)
+            return if (parts[0] == "<token>") {
```
Comment (charleskorn) [inline libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt:48]: This might be even clearer as a `when`.

File: libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt [hunk 5]
```diff
@@ -90,10 +90,10 @@ data class HelperBasedCredentialsSource(val helperName: String, val serverAddres
 
         val serverUrl = parsed.getStringMember("ServerURL")
 
-        if (username == "<token>") {
```
Comment (charleskorn) [inline libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt:93]: Ditto: this might be even clearer as a `when`.

**Review by charleskorn — CHANGES_REQUESTED** [review]
Thanks for doing this! There are a couple of places where the fixes (at least IMHO) make the code harder to read, so I'd prefer to keep the original expressions, but otherwise this looks good to me.

File: app/src/main/kotlin/batect/cli/commands/HelpCommand.kt (hunk 1 above)
Comment (sschuberth) [reply app/src/main/kotlin/batect/cli/commands/HelpCommand.kt:62]: I've mostly revered the change, but still applied the "'when' with subject should be used" suggestion from IntelliJ. Let's see if you like it.

File: app/src/main/kotlin/batect/cli/options/OptionParser.kt (hunk 2 above)
Comment (sschuberth) [reply app/src/main/kotlin/batect/cli/options/OptionParser.kt:54]: I've mostly reverted the change, but still applied the `'Return' can be lifted out of 'if'` suggestion. I've *not* applied the also present `If-Null return/break/... foldable to '?:'` suggestion, as I believe that's the bit which makes it less readable for you. Feel free to play around with it, though, as there might be a compromise, like

        val option = optionNames[argName]
            ?: return if (argName.startsWith("-")) {
                OptionParsingResult.InvalidOption("Invalid option '$argName'. Run './batect --help' for a list of valid options.")
            } else {
                OptionParsingResult.NoOption
            }

        return option.parse(args.drop(currentIndex))

File: app/src/main/kotlin/batect/proxies/ProxyEnvironmentVariablesProvider.kt (hunk 3 above)
Comment (sschuberth) [reply app/src/main/kotlin/batect/proxies/ProxyEnvironmentVariablesProvider.kt:37]: Done manually.

File: libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt (hunk 4 above)
Comment (sschuberth) [reply libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt:48]: Personally, I avoid using `when` if there only one condition and an `else`. To me, that's the typical `if` use-case, and `when`'s power lies in using a subject, which usually does not make sense in this case.

File: libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt (hunk 5 above)
Comment (sschuberth) [reply libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt:93]: Please check if you agree to my rationale above.

Comment (sschuberth) [reply libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt:93]: I actually still gave it a try, and indeed found it to result in more compact code, so done (in both places).

**Review by charleskorn — APPROVED** [review]
Looks good to me, thanks again for submitting these improvements. I'll merge this once the build passes.
